### PR TITLE
add_task_to_user

### PR DESCRIPTION
**Pull request by trungnn142**

### CODE REVIEW

File: app/views/user_subjects/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <fieldset class="fsStyle">
+      <legend class="legendStyle">
+        <span class="label label-info"><%= t "supervisor.subject.show.task" %></span>
+      </legend>
+      <%= form_for @user_subject do |f| %>
+        <div class="row">
+          <%= f.collection_check_boxes :task_ids, @tasks, :id, :name do |b| %>
+            <div class="col-lg-2">
+              <% unless task_finished? b.object.id, @user_id %>
+                  <div class="thumbnail text-center">
```
Comment (hellovietnam93) [inline app/views/user_subjects/show.html.erb]: ident

File: app/views/user_subjects/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    <h4><%= b.object.name %></h4>
+                    <p><%= b.check_box %></p>
+                  </div>
+              <% else %>
+                  <div class="thumbnail thumbnail-disable text-center">
```
Comment (hellovietnam93) [inline app/views/user_subjects/show.html.erb]: ident

File: app/controllers/user_subjects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def update
+    if @user_subject.update user_subject_params
+      @user_subject.user_tasks.each do |user_task|
+        user_task.update_attributes user_id: @user_subject.user_id
+      end
+      flash[:success] = t "application.flash.user_subject_updated"
+      redirect_to @user_subject
+    end
```
Comment (ThuBM) [inline app/controllers/user_subjects_controller.rb]: update fails?

File: app/controllers/user_subjects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class UserSubjectsController < ApplicationController
+  before_action :load_user_subject
```
Comment (ThuBM) [inline app/controllers/user_subjects_controller.rb]: only

File: app/views/user_subjects/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+              <% unless task_finished? b.object.id, @user_id %>
+                <div class="thumbnail text-center">
+                  <h4><%= b.object.name %></h4>
+                  <p><%= b.check_box %></p>
+                </div>
+              <% else %>
+                <div class="thumbnail thumbnail-disable text-center">
+                  <h4><%= b.object.name %></h4>
+                  <p><%= b.check_box disabled: true %></p>
+                  <%= hidden_field_tag :"user_subject[task_ids][]", b.object.id %>
```
Comment (ThuBM) [inline app/views/user_subjects/show.html.erb]: remove :

File: config/routes.rb [hunk 6]
```diff
@@ -16,5 +16,6 @@
   end
 
   resources :users
+  resources :user_subjects, only: [:show, :edit, :update]
```
Comment (ThuBM) [inline config/routes.rb]: resource

**Comment by ThuBM** [comment]
OK

File: app/controllers/user_subjects_controller.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+  def show
+    @subject = @user_subject.subject
+    @tasks = @subject.tasks
+    @user_id = @user_subject.user_id
+  end
+
+  def update
+    if @user_subject.update user_subject_params
+      @user_subject.user_tasks.each do |user_task|
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: call_back

File: app/helpers/user_subjects_helper.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+module UserSubjectsHelper
+  def task_finished? task_id, user_id
+    UserTask.find_by task_id: task_id, user_id: user_id
+  end
+
+  def user_subject_finished? user_subject
+    user_subject.user_tasks.count < user_subject.subject.tasks.count ? false : true
```
Comment (dieunb) [inline app/helpers/user_subjects_helper.rb]: simple `user_subject.user_tasks.count < user_subject.subject.tasks.count`

File: app/helpers/user_subjects_helper.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+module UserSubjectsHelper
+  def task_finished? task_id, user_id
+    UserTask.find_by task_id: task_id, user_id: user_id
+  end
+
+  def user_subject_finished? user_subject
+    user_subject.subject.tasks.count == user_subject.user_tasks.count
+  end
+
+  def percent_complete_subject user_subject
```
Comment (dieunb) [inline app/helpers/user_subjects_helper.rb]: http://apidock.com/rails/ActionView/Helpers/NumberHelper/number_to_percentage

File: app/models/user_subject.rb [hunk 10]
```diff
@@ -22,4 +23,10 @@ def create_subject_completed_log
       create_activity_log subject_id, "subject_completed", course.id, id
     end
   end
+
+  def update_user_id_to_user_tasks
+    self.user_tasks.each do |user_task|
+        user_task.update_attributes user_id: self.user_id
```
Comment (dieunb) [inline app/models/user_subject.rb]: indent

File: app/views/user_subjects/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+<div class="row">
+  <div class="col-lg-12">
+    <p><%= @subject.instruction %></p><br />
+    <fieldset class="fsStyle">
+      <legend class="legendStyle">
+        <span class="label label-primary"><%= t "supervisor.subject.show.task" %></span>
+      </legend>
+      <%= form_for @user_subject, url: subject_user_subject_path(@subject) do |f| %>
+        <div class="row">
+          <% @tasks.each do |task| %>
```
Comment (dieunb) [inline app/views/user_subjects/show.html.erb]: `field_for`

File: app/controllers/user_subjects_controller.rb [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  include UserSubjectsHelper
+  before_action :load_user_subject, only: [:show, :update]
+  before_action :authenticate_user!
+
+  def show
+    @subject = @user_subject.subject
+    @tasks = @subject.tasks
+    @user_id = @user_subject.user_id
+    @tasks.each do |task|
+      unless task_finished? task.id, @user_id
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: `find_or_initialize_by`

**Pull request merged**